.circleci/config.yml

```diff
@@ -22,41 +22,46 @@ executors:
 commands:
   restore-cargo-cache:
     steps:
-      # Restore cargo cache before installing anything with the cargo command (ie cargo install ...)
       - restore_cache:
-          name: Restore cargo cache
-          keys:
-            - cargo-{{ checksum "Cargo.lock" }}
+          name: Restore cargo registry cache
+          key: cargo-{{ checksum "Cargo.lock" }}
+          # Don't use fallback key to prevent registry cache from growing indefinitely.
       - run:
           name: Install sccache
+          # 500 MB is the recommended soft cap for a cache on circleci.
+          # Large integration tests override this default.
           command: |
-            export SCCACHE_VERSION='v0.5.3'
-            ls ~/.cargo/bin/sccache || curl -L https://github.com/mozilla/sccache/releases/download/$SCCACHE_VERSION/sccache-$SCCACHE_VERSION-x86_64-unknown-linux-musl.tar.gz | tar -xOz sccache-$SCCACHE_VERSION-x86_64-unknown-linux-musl/sccache > ~/.cargo/bin/sccache && chmod +x ~/.cargo/bin/sccache
-            # This configures Rust to use sccache.
-            echo 'export "RUSTC_WRAPPER"="sccache"' >> $BASH_ENV
-            sccache --version
+            SCCACHE_VERSION='v0.5.4'
+            ls ~/.cargo/bin/sccache \
+              || curl -L https://github.com/mozilla/sccache/releases/download/$SCCACHE_VERSION/sccache-$SCCACHE_VERSION-x86_64-unknown-linux-musl.tar.gz \
+                | tar -xOz sccache-$SCCACHE_VERSION-x86_64-unknown-linux-musl/sccache \
+                > ~/.cargo/bin/sccache \
+              && chmod +x ~/.cargo/bin/sccache
+            echo 'export RUSTC_WRAPPER=~/.cargo/bin/sccache' >> $BASH_ENV
+            echo 'export SCCACHE_CACHE_SIZE=500M' >> $BASH_ENV
       - restore_cache:
           name: Restore sccache cache
-          key: sccache-cache-{{ .Environment.CIRCLE_JOB }}
+          keys: # This can use fallback due to size limit.
+            - sccache-cache-{{ .Environment.CIRCLE_JOB }}-{{ .Branch }}-
+            - sccache-cache-{{ .Environment.CIRCLE_JOB }}-
```

Comment:
Why are the `-` needed at the end?

Reply:
Doesn't matter if they are there or not. But I think it shows more clearly that they are trying to match a prefix.